test-utils/test-framework/src/framework.rs

```diff
@@ -0,0 +1,116 @@
+use actix_rt;
+use anyhow;
+use std::future::Future;
+use std::path::{Path, PathBuf};
+use std::sync::{Arc, Mutex};
+
+use crate::YagnaMock;
+
+pub use macros::prepare_test_dir;
+pub use ya_framework_macro::framework_test;
+
+#[derive(Clone)]
+pub struct YagnaFramework {
+    inner: Arc<Mutex<YagnaNetworkImpl>>,
+    test_dir: PathBuf,
```

Comment:
You could use `tempdir::TempDir` for that. It will allow to run tests in parallel- every testdir will have it's own directory which won't collide

Reply:
1. This directory contains yagna datadirs. In case of integration test, there is no chance that tests are self-explanatory. You are running whole yagna. It is not easy to debug it, without having datadir. Logs are included in too.
2. Running in parallel will be theoretically possible, because test name is rather unique. Maybe some name mangling will be necessary. But `framework_test` macro is using `serial_test` to avoid parallel execution. Integrations tests are rather too heavy to run in parallel

Reply:
Ad 1 & Ad 2:
I don't agree that if test is integration- than it should automatically be complicated and it will test multiple things at once.
When tests are stable, if you must debug some complicated test it means for me that there is lack of other-simpler case test. Because if simpler test passes- fail is because of functionality which is added by this more complex one. If simpler fails- it means that it doesn't matter that complex one is failing as problem lays in simple case.

Of course when tests are very slow, it may be difficult to run many small-case tests. But then we come to second point that we can run them in parallel. I think there is some misconception here:
We design framework where tests can't be ran in parallel -> so running many tests becomes slow -> so we naturally will dive into making big & complicated tests and not many simpler ones -> we will end up in tests which will be hard to understand and maintain. When something will break it will require many time to debug

Reply:
Tests don't have to be complicated. Running yagna makes it complicated in itself.
Running tests in parallel isn't solution for long running tests, because it only can speed up as many times as many cores you have. And running to many yagnas in the background can harm their performance and we can end up with flaky tests like in goth (which we will end up anyway, but we can try to minimize it)